pkg/aws/client.go

```diff
@@ -611,13 +613,53 @@ func (c *awsClient) FilterVPCsPrivateSubnets(subnets []*ec2.Subnet) ([]*ec2.Subn
 		}
 	}
 
+	// Temporary filter until local-zone and wave-length zones are supported by HCP
+	if isHostedCp {
+		privateSubnets, err = c.FilterSubnetsWithStandardAvailabilityZones(privateSubnets)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	if len(privateSubnets) < 1 {
 		return nil, fmt.Errorf("Failed to find private subnets associated with VPC '%s'", *subnets[0].VpcId)
 	}
 
 	return privateSubnets, nil
 }
 
+func (c *awsClient) FilterSubnetsWithStandardAvailabilityZones(subnets []*ec2.Subnet) ([]*ec2.Subnet, error) {
+	filteredSubnets := []*ec2.Subnet{}
+	subnetAvailabilityZonesList := []*string{}
+	standardAvailabilityZonesMap := map[string]bool{}
+
+	// list of availability zones to pass as input
+	for _, subnet := range subnets {
+		subnetAvailabilityZonesList = append(subnetAvailabilityZonesList, subnet.AvailabilityZone)
+	}
+
+	describeAvailabilityZonesOutput, err := c.ec2Client.DescribeAvailabilityZones(
+		&ec2.DescribeAvailabilityZonesInput{ZoneNames: subnetAvailabilityZonesList})
+	if err != nil {
+		return filteredSubnets, err
+	}
+
+	for _, availabilityZone := range describeAvailabilityZonesOutput.AvailabilityZones {
+		if *availabilityZone.ZoneType == standardZone {
```

Comment:
any risk to do this instead of filtering out the 2 types we don't want? It means that if there is another type we risk missing that too

Reply:
As of now, the only 3 types of az types exposed via AWS SDK are,
"availability-zone (standard)", "local-zone", "wavelength-zone".
These az types seem to be created and classified based on how close they are to edge locations.

I am thinking that we on ly expose to customers, subnets / az types that are already confirmed / tested and compatible with Openshift, vs. having the possibility of exposing new subnets / az types if they become readily available via the SDK, without our knowledge - if we don't keep track.

Reply:
thanks, makes sense